test/end-to-end/core.sh

```diff
@@ -184,6 +184,11 @@ os::cmd::expect_success_and_text "cat '${LOG_DIR}/kubectl-with-token.log'" 'Usin
 os::cmd::expect_success_and_text "cat '${LOG_DIR}/kubectl-with-token.log'" 'kubectl-with-token'
 
 echo "[INFO] Testing deployment logs and failing pre and mid hooks ..."
+# test hook selectors
+os::cmd::expect_success "oc create -f ${OS_ROOT}/test/fixtures/complete-dc-hooks.yaml"
+os::cmd::try_until_text 'oc get pods -l openshift.io/deployer-pod.type=hook-pre  -o jsonpath={.items[*].status.phase}' 'Succeeded'
```

Comment:
With `.items[*].status.phase` you will get a list of phases, and your assert is that at least one of these phases is `Succeeded` -- is that what we want? At least one pod has succeeded? Do we want to know that all of them have?